### PR TITLE
test/evp_test.c: don't misuse pkey_test_ctrl() in mac_test_run()

### PR DESCRIPTION
**Pull request by levitte**
pkey_test_ctrl() was designed for parsing values, not for using in
test runs.  Relying on its returned value when it returned 1 even for
control errors made it particularly useless for mac_test_run().

Here, it gets replaced with a MAC specific control function, that
parses values the same way but is designed for use in a _run() rather
than a _parse() function.

This uncovers a SipHash test with an invalid control that wasn't
caught properly.  After all, that stanza is supposed to test that
invalid control values do generate an error.  Now we catch that.

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/recipes/30-test_evp_data/evpmac.txt [hunk 1]
```diff
@@ -159,6 +159,7 @@ Ctrl = digestsize:13
 Key = 000102030405060708090A0B0C0D0E0F
 Input = 000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E
 Output = 5150d1772f50834a503e069a973fbd7c
```
Comment (t-j-h) [inline test/recipes/30-test_evp_data/evpmac.txt]: Why is this showing output if it is meant to be catching an error?

Comment (levitte) [reply test/recipes/30-test_evp_data/evpmac.txt]: It doesn't, so the Input and Output lines are actually unnecessary...  unless removing them makes the parser fail, which I hadn't verified.  Coming up.

Comment (levitte) [reply test/recipes/30-test_evp_data/evpmac.txt]: (I have some rebase work to do anyway)

Comment (levitte) [reply test/recipes/30-test_evp_data/evpmac.txt]: Fixed.  The key must remain, though...

Comment (t-j-h) [reply test/recipes/30-test_evp_data/evpmac.txt]: Showing Output seems wrong - Input I can understand - but if the code doesn't make it to that part then I would expect to see zero length Output if the field has to be present.

**Comment by levitte** [comment]
Ok, CIs seem to be happy

**Review by t-j-h — APPROVED** [review]

**Comment by levitte** [comment]
Merged.

master:
ce5d64c79c4d809ece8fe28a5b62915467a1c0e1 test/evp_test.c: don't misuse pkey_test_ctrl() in mac_test_run()

1.1.1:
d308458ef138dfbe925203b5cb01a015f0aa93ac test/evp_test.c: don't misuse pkey_test_ctrl() in mac_test_run()

I forgot that the test programs in 1.1.0 is very different.  Thinking that nah, not worth it, especially since we don't even have the algorithm which' tests are affected there, so...

**Pull request closed without merging**